### ISSUE TITLE
Different `train` stage `metrics` and `params` paths relative to Git repo root

### ISSUE THREAD

**Issue by aschuh-hf** [iterative/dvc#9860]
I just had a long running queued experiment finish. The results of it are shown in the Experiments table. However, the paths of the `params.yaml` and `metrics.json` file have changed and are different for previously run experiments and this latest one. I haven't changed my `dvc.yaml` file. I have updated my DVC VS Code Extension today, however.

<img width="618" alt="Screen Shot 2023-08-17 at 8 58 42 PM" src="https://github.com/iterative/vscode-dvc/assets/77496589/025566c3-1f8f-4f18-ad43-1a432680e136">

DVC VS Code Extension v1.0.43

```
$ dvc doctor
DVC version: 3.9.1 (conda)
--------------------------
Platform: Python 3.10.6 on Linux-3.10.0-1127.8.2.el7.x86_64-x86_64-with-glibc2.17
Subprojects:
        dvc_data = 2.8.1
        dvc_objects = 0.25.0
        dvc_render = 0.5.3
        dvc_task = 0.3.0
        scmrepo = 1.2.1
Supports:
        http (aiohttp = 3.8.5, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.5, aiohttp-retry = 2.8.3),
        s3 (s3fs = 2023.6.0, boto3 = 1.26.76)
Config:
        Global: /home/aschuh/.config/dvc
        System: /etc/xdg/dvc
Cache types: hardlink, symlink
Cache directory: xfs on /dev/sda1
Caches: local
Remotes: s3, s3
Workspace directory: xfs on /dev/sda1
Repo: dvc (subdir), git
Repo.site_cache_dir: /var/tmp/dvc/repo/764b1d03f8b416f032df6c50c3809e7f
```

**Comment by dberenbaum** [comment]
@aschuh-hf Do you know how to reproduce it? I'm unable to reproduce by running a new experiment from https://github.com/iterative/monorepo-example:

<img width="245" alt="Screenshot 2023-08-21 at 8 33 24 AM" src="https://github.com/iterative/dvc/assets/2308172/c96ad036-71a0-4d7e-91ab-5f15b48d0b18">

<details>

<summary>`dvc exp show --json` output</summary>

```json
$ dvc exp show --json
[
  {
    "rev": "workspace",
    "name": null,
    "data": {
      "rev": "workspace",
      "timestamp": null,
      "params": {
        "results/train/params.yaml": {
          "data": {
            "model": "DynamicUnet",
            "batch_size": 8,
            "batch_per_epoch": 8,
            "frozen": false,
            "frozen_idx": 0,
            "transforms": "[Pipeline: PILBase.create, Pipeline: partial -> PILBase.create]"
          }
        },
        "params.yaml": {
          "data": {
            "base": {
              "random_seed": 42
            },
            "data_split": {
              "test_pct": 0.15
            },
            "train": {
              "valid_pct": 0.1,
              "arch": "shufflenet_v2_x2_0",
              "img_size": 256,
              "batch_size": 8,
              "fine_tune_args": {
                "epochs": 8,
                "base_lr": 0.001
              }
            },
            "evaluate": {
              "n_samples_to_save": 10
            }
          }
        }
      },
      "metrics": {
        "results/evaluate/metrics.json": {
          "data": {
            "dice_multi": 0.6948640039028317
          }
        },
        "results/train/metrics.json": {
          "data": {
            "train": {
              "loss": 0.07587292045354843
            },
            "eval": {
              "loss": 0.049336470663547516
            },
            "dice_multi": 0.7944949522123549,
            "step": 8
          }
        }
      },
      "deps": {
        "data/pool_data": {
          "hash": "14d187e749ee5614e105741c719fa185.dir",
          "size": 18999874,
          "nfiles": 183
        },
        "src/data_split.py": {
          "hash": "e25f97724df2d2dfdfd05e228b788a58",
          "size": 1011,
          "nfiles": null
        },
        "data/train_data": {
          "hash": "a14914527fa6a82f91911013614d4068.dir",
          "size": 15530783,
          "nfiles": 148
        },
        "src/train.py": {
          "hash": "ef9d70ee318c6b345ba974af4d2af4e4",
          "size": 2088,
          "nfiles": null
        },
        "data/test_data": {
          "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
          "size": 3462943,
          "nfiles": 34
        },
        "models/model.pkl": {
          "hash": "de9d1504eb4b0b8d5d64719fa81e11d7",
          "size": 203255,
          "nfiles": null
        },
        "src/evaluate.py": {
          "hash": "e2d5d64f6064e95b7661b239b8b20626",
          "size": 3364,
          "nfiles": null
        }
      },
      "outs": {
        "data/test_data": {
          "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
          "size": 3462943,
          "nfiles": 34,
          "use_cache": true,
          "is_data_source": false
        },
        "data/train_data": {
          "hash": "a14914527fa6a82f91911013614d4068.dir",
          "size": 15530783,
          "nfiles": 148,
          "use_cache": true,
          "is_data_source": false
        },
        "models/model.pkl": {
          "hash": "de9d1504eb4b0b8d5d64719fa81e11d7",
          "size": 203255,
          "nfiles": null,
          "use_cache": true,
          "is_data_source": false
        },
        "data/pool_data": {
          "hash": "14d187e749ee5614e105741c719fa185.dir",
          "size": 18999874,
          "nfiles": 183,
          "use_cache": true,
          "is_data_source": true
        }
      },
      "meta": {}
    },
    "error": null,
    "experiments": null
  },
  {
    "rev": "a1ee98d75055cae149e5f40678739509f1683653",
    "name": "develop",
    "data": {
      "rev": "a1ee98d75055cae149e5f40678739509f1683653",
      "timestamp": "2023-06-07T19:01:01",
      "params": {
        "results/train/params.yaml": {
          "data": {
            "model": "DynamicUnet",
            "batch_size": 8,
            "batch_per_epoch": 8,
            "frozen": false,
            "frozen_idx": 0,
            "transforms": "[Pipeline: PILBase.create, Pipeline: partial -> PILBase.create]"
          }
        },
        "params.yaml": {
          "data": {
            "base": {
              "random_seed": 42
            },
            "data_split": {
              "test_pct": 0.15
            },
            "train": {
              "valid_pct": 0.1,
              "arch": "shufflenet_v2_x2_0",
              "img_size": 256,
              "batch_size": 8,
              "fine_tune_args": {
                "epochs": 8,
                "base_lr": 0.01
              }
            },
            "evaluate": {
              "n_samples_to_save": 10
            }
          }
        }
      },
      "metrics": {
        "results/evaluate/metrics.json": {
          "data": {
            "dice_multi": 0.8667462817565745
          }
        },
        "results/train/metrics.json": {
          "data": {
            "train": {
              "loss": 0.03061583824455738
            },
            "eval": {
              "loss": 0.021420806646347046
            },
            "dice_multi": 0.8728481883947294,
            "step": 8
          }
        }
      },
      "deps": {
        "data/pool_data": {
          "hash": "14d187e749ee5614e105741c719fa185.dir",
          "size": 18999874,
          "nfiles": 183
        },
        "src/data_split.py": {
          "hash": "e25f97724df2d2dfdfd05e228b788a58",
          "size": 1011,
          "nfiles": null
        },
        "data/train_data": {
          "hash": "a14914527fa6a82f91911013614d4068.dir",
          "size": 15530783,
          "nfiles": 148
        },
        "src/train.py": {
          "hash": "ef9d70ee318c6b345ba974af4d2af4e4",
          "size": 2088,
          "nfiles": null
        },
        "data/test_data": {
          "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
          "size": 3462943,
          "nfiles": 34
        },
        "models/model.pkl": {
          "hash": "cf0c77f4d1a1b222e39f83908959449a",
          "size": 201335,
          "nfiles": null
        },
        "src/evaluate.py": {
          "hash": "e2d5d64f6064e95b7661b239b8b20626",
          "size": 3364,
          "nfiles": null
        }
      },
      "outs": {
        "data/test_data": {
          "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
          "size": 3462943,
          "nfiles": 34,
          "use_cache": true,
          "is_data_source": false
        },
        "data/train_data": {
          "hash": "a14914527fa6a82f91911013614d4068.dir",
          "size": 15530783,
          "nfiles": 148,
          "use_cache": true,
          "is_data_source": false
        },
        "models/model.pkl": {
          "hash": "cf0c77f4d1a1b222e39f83908959449a",
          "size": 201335,
          "nfiles": null,
          "use_cache": true,
          "is_data_source": false
        },
        "data/pool_data": {
          "hash": "14d187e749ee5614e105741c719fa185.dir",
          "size": 18999874,
          "nfiles": 183,
          "use_cache": true,
          "is_data_source": true
        }
      },
      "meta": {}
    },
    "error": null,
    "experiments": [
      {
        "revs": [
          {
            "rev": "f99c39a06ce49959affe9147df18aa0b4a54072e",
            "name": "unlet-zigs",
            "data": {
              "rev": "f99c39a06ce49959affe9147df18aa0b4a54072e",
              "timestamp": "2023-08-21T08:33:14",
              "params": {
                "results/train/params.yaml": {
                  "data": {
                    "model": "DynamicUnet",
                    "batch_size": 8,
                    "batch_per_epoch": 8,
                    "frozen": false,
                    "frozen_idx": 0,
                    "transforms": "[Pipeline: PILBase.create, Pipeline: partial -> PILBase.create]"
                  }
                },
                "params.yaml": {
                  "data": {
                    "base": {
                      "random_seed": 42
                    },
                    "data_split": {
                      "test_pct": 0.15
                    },
                    "train": {
                      "valid_pct": 0.1,
                      "arch": "shufflenet_v2_x2_0",
                      "img_size": 256,
                      "batch_size": 8,
                      "fine_tune_args": {
                        "epochs": 8,
                        "base_lr": 0.001
                      }
                    },
                    "evaluate": {
                      "n_samples_to_save": 10
                    }
                  }
                }
              },
              "metrics": {
                "results/evaluate/metrics.json": {
                  "data": {
                    "dice_multi": 0.6948640039028317
                  }
                },
                "results/train/metrics.json": {
                  "data": {
                    "train": {
                      "loss": 0.07587292045354843
                    },
                    "eval": {
                      "loss": 0.049336470663547516
                    },
                    "dice_multi": 0.7944949522123549,
                    "step": 8
                  }
                }
              },
              "deps": {
                "data/pool_data": {
                  "hash": "14d187e749ee5614e105741c719fa185.dir",
                  "size": 18999874,
                  "nfiles": 183
                },
                "src/data_split.py": {
                  "hash": "e25f97724df2d2dfdfd05e228b788a58",
                  "size": 1011,
                  "nfiles": null
                },
                "data/train_data": {
                  "hash": "a14914527fa6a82f91911013614d4068.dir",
                  "size": 15530783,
                  "nfiles": 148
                },
                "src/train.py": {
                  "hash": "ef9d70ee318c6b345ba974af4d2af4e4",
                  "size": 2088,
                  "nfiles": null
                },
                "data/test_data": {
                  "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
                  "size": 3462943,
                  "nfiles": 34
                },
                "models/model.pkl": {
                  "hash": "de9d1504eb4b0b8d5d64719fa81e11d7",
                  "size": 203255,
                  "nfiles": null
                },
                "src/evaluate.py": {
                  "hash": "e2d5d64f6064e95b7661b239b8b20626",
                  "size": 3364,
                  "nfiles": null
                }
              },
              "outs": {
                "data/test_data": {
                  "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
                  "size": 3462943,
                  "nfiles": 34,
                  "use_cache": true,
                  "is_data_source": false
                },
                "data/train_data": {
                  "hash": "a14914527fa6a82f91911013614d4068.dir",
                  "size": 15530783,
                  "nfiles": 148,
                  "use_cache": true,
                  "is_data_source": false
                },
                "models/model.pkl": {
                  "hash": "de9d1504eb4b0b8d5d64719fa81e11d7",
                  "size": 203255,
                  "nfiles": null,
                  "use_cache": true,
                  "is_data_source": false
                },
                "data/pool_data": {
                  "hash": "14d187e749ee5614e105741c719fa185.dir",
                  "size": 18999874,
                  "nfiles": 183,
                  "use_cache": true,
                  "is_data_source": true
                }
              },
              "meta": {}
            },
            "error": null,
            "experiments": null
          }
        ],
        "executor": null,
        "name": "unlet-zigs"
      },
      {
        "revs": [
          {
            "rev": "83fb2b0aa2c63980ff97aa7a0379c5ef713e1901",
            "name": "manky-veil",
            "data": {
              "rev": "83fb2b0aa2c63980ff97aa7a0379c5ef713e1901",
              "timestamp": "2023-08-21T08:30:59",
              "params": {
                "results/train/params.yaml": {
                  "data": {
                    "model": "DynamicUnet",
                    "batch_size": 8,
                    "batch_per_epoch": 8,
                    "frozen": false,
                    "frozen_idx": 0,
                    "transforms": "[Pipeline: PILBase.create, Pipeline: partial -> PILBase.create]"
                  }
                },
                "params.yaml": {
                  "data": {
                    "base": {
                      "random_seed": 42
                    },
                    "data_split": {
                      "test_pct": 0.15
                    },
                    "train": {
                      "valid_pct": 0.1,
                      "arch": "shufflenet_v2_x2_0",
                      "img_size": 256,
                      "batch_size": 8,
                      "fine_tune_args": {
                        "epochs": 8,
                        "base_lr": 0.01
                      }
                    },
                    "evaluate": {
                      "n_samples_to_save": 10
                    }
                  }
                }
              },
              "metrics": {
                "results/evaluate/metrics.json": {
                  "data": {
                    "dice_multi": 0.8667462817565745
                  }
                },
                "results/train/metrics.json": {
                  "data": {
                    "train": {
                      "loss": 0.03061583824455738
                    },
                    "eval": {
                      "loss": 0.021420806646347046
                    },
                    "dice_multi": 0.8728481883947294,
                    "step": 8
                  }
                }
              },
              "deps": {
                "data/pool_data": {
                  "hash": "14d187e749ee5614e105741c719fa185.dir",
                  "size": 18999874,
                  "nfiles": 183
                },
                "src/data_split.py": {
                  "hash": "e25f97724df2d2dfdfd05e228b788a58",
                  "size": 1011,
                  "nfiles": null
                },
                "data/train_data": {
                  "hash": "a14914527fa6a82f91911013614d4068.dir",
                  "size": 15530783,
                  "nfiles": 148
                },
                "src/train.py": {
                  "hash": "ef9d70ee318c6b345ba974af4d2af4e4",
                  "size": 2088,
                  "nfiles": null
                },
                "data/test_data": {
                  "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
                  "size": 3462943,
                  "nfiles": 34
                },
                "models/model.pkl": {
                  "hash": "cf0c77f4d1a1b222e39f83908959449a",
                  "size": 201335,
                  "nfiles": null
                },
                "src/evaluate.py": {
                  "hash": "e2d5d64f6064e95b7661b239b8b20626",
                  "size": 3364,
                  "nfiles": null
                }
              },
              "outs": {
                "data/test_data": {
                  "hash": "1f0a1e034255ad40c65cf4858ed404cd.dir",
                  "size": 3462943,
                  "nfiles": 34,
                  "use_cache": true,
                  "is_data_source": false
                },
                "data/train_data": {
                  "hash": "a14914527fa6a82f91911013614d4068.dir",
                  "size": 15530783,
                  "nfiles": 148,
                  "use_cache": true,
                  "is_data_source": false
                },
                "models/model.pkl": {
                  "hash": "cf0c77f4d1a1b222e39f83908959449a",
                  "size": 201335,
                  "nfiles": null,
                  "use_cache": true,
                  "is_data_source": false
                },
                "data/pool_data": {
                  "hash": "14d187e749ee5614e105741c719fa185.dir",
                  "size": 18999874,
                  "nfiles": 183,
                  "use_cache": true,
                  "is_data_source": true
                }
              },
              "meta": {}
            },
            "error": null,
            "experiments": null
          }
        ],
        "executor": null,
        "name": "manky-veil"
      }
    ]
  }
]
```

</details>

**Comment by aschuh-hf** [comment]
I don't think I could reproduce this myself, unfortunately. Not sure what triggered it either. I didn't have the same issue with subsequent / more recent experiment runs.

Maybe the update of DVC did resolve this from occurring in the first place, even if it still displayed the same symptom?

**Comment by dberenbaum** [comment]
No problem. I'll close, but feel free to reopen if it happens again.

**Comment by aschuh-hf** [comment]
I just encountered this issue again with 2 out of 11 experiments that I ran in parallel. The experiments finished at different times, but it may be that 2-3 experiments finished about the same time with overlap of the execution of the Git operations to finalize the experiments given that these operations in our mono-repo take a bit longer. I wonder if this could have anything to do with it?

I now used

```
dvc exp rename <exp> <exp>-needs-fixup
dvc exp apply <exp>-needs-fixup
dvc exp save -n <exp>
```

to create a new experiment with correct `metrics` and `params` paths, but left the original experiment showing the issue under the different name around in the hope this can help debug the issue. Let me know what information would be useful to figure out the cause for this.

See below a screenshot of the Experiments table in VS Code after those steps. I marked the affected rows by selecting them for plotting, and starred the newly saved experiments created using the above commands.

<img width="2780" alt="Screen Shot 2023-10-07 at 12 23 23 PM" src="https://github.com/iterative/dvc/assets/77496589/fcd1655d-1265-4da9-9e17-44c62f495996">

**Comment by mattseddon** [comment]
@dberenbaum, when this shows up, it causes issues for the extension's file watchers.

**Comment by skshetry** [comment]
@aschuh-hf, can you try removing `.dvc/tmp/exps/cache` directory and retry `dvc exp show`?

We have fixed stuff like this in [3.20.0](https://github.com/iterative/dvc/releases/tag/3.20.0), so it may be because of the cache.

**Comment by aschuh-hf** [comment]
@skshetry That's it! Deleting the `.dvc/tmp/exps/cache` directory fixed it while using DVC 3.22.1. Thank you!
